cmd/containerd-shim-runhcs-v1/task.go

```diff
@@ -86,6 +89,30 @@ type shimTask interface {
 	// If the host is hypervisor isolated and this task owns the host additional
 	// metrics on the UVM may be returned as well.
 	Stats(ctx context.Context) (*stats.Statistics, error)
+	// Update updates a task's container
+	Update(ctx context.Context, req *task.UpdateTaskRequest) error
+}
+
+func verifyTaskUpdateResourcesType(data interface{}) error {
+	switch data.(type) {
+	case *specs.WindowsResources:
+	case *specs.LinuxResources:
+	default:
+		return errors.New("update resources must be of type *WindowsResources or *LinuxResources")
+	}
+	return nil
+}
+
+func normalizeContainerMemorySize(ctx context.Context, cid string, requestedSize uint64) uint64 {
```

Comment:
It would help readability if we included units in the variable/field names:
- `requestedSizeMB`
- `actualMB`
- `"requestedMB"`
- `"assignedMB"`